src/openms/source/ANALYSIS/ID/FalseDiscoveryRate.cpp

```diff
@@ -283,6 +317,28 @@ namespace OpenMS
         map<double, double> score_to_fdr;
         calculateFDRs_(score_to_fdr, target_scores, decoy_scores, q_value, higher_score_better);
 
+        // calculate peptide FDR
+        vector<double> decoy_peptide_scores, target_peptide_scores;
+        for (const auto& ps : peptide_to_best_decoy_score)
+        {
+          decoy_peptide_scores.push_back(ps.second);
+        }
+        for (const auto& ps : peptide_to_best_target_score)
+        {
+          target_peptide_scores.push_back(ps.second);
+        }      
+        map<double, double> score_to_peptide_fdr;
+        calculateFDRs_(score_to_peptide_fdr, target_peptide_scores, decoy_peptide_scores, q_value, higher_score_better);
```

Comment:
I think applyBasic is faster and more correct

Reply:
puhhh this is how it i done for the others. maybe make a note or open an issue? I think FDR is  one of the classes that should get a simple API - there are now just so many ways to do things I completely lost track what is the best way to do things

Reply:
I just kept it for backwards compatibility. If someone can explain and comment the code, it can stay. It also needs a complexity analysis since it is a very central part. Otherwise, if I don't understand it, I would not use it.

Reply:
For now it is fine.